core/templates/dev/head/pages/exploration_editor/ExplorationDataService.js

```diff
@@ -103,31 +105,37 @@ oppia.factory('ExplorationDataService', [
           // changes applied. This makes a force-refresh necessary when changes
           // are discarded, otherwise the exploration-with-draft-changes
           // (which is cached here) will be reused.
-          return (
+          return Promise.all([
             EditableExplorationBackendApiService.fetchApplyDraftExploration(
-              explorationId).then(function(response) {
-              $log.info('Retrieved exploration data.');
-              $log.info(response);
-              draftChangeListId = response.draft_change_list_id;
-              explorationData.data = response;
-              var draft = LocalStorageService.getExplorationDraft(
-                explorationId);
-              if (draft) {
-                if (draft.isValid(draftChangeListId)) {
-                  var changeList = draft.getChanges();
-                  explorationData.autosaveChangeList(changeList, function() {
-                    // A reload is needed so that the changelist just saved is
-                    // loaded as opposed to the exploration returned by this
-                    // response.
-                    $window.location.reload();
-                  });
-                } else {
-                  errorCallback(explorationId, draft.getChanges());
-                }
+              explorationId),
+            StateTopAnswersStatsBackendApiService.fetchStats(explorationId),
+          ]).then(function(promisedValues) {
+            var explorationDataResponse = promisedValues[0];
+            var stateTopAnswersStatsResponse = promisedValues[1];
+            $log.info('Retrieved exploration data.');
+            $log.info(explorationDataResponse);
+            $log.info(stateTopAnswersStatsResponse);
+            draftChangeListId = explorationDataResponse.draft_change_list_id;
+            explorationData.data = explorationDataResponse;
+            var draft = LocalStorageService.getExplorationDraft(
+              explorationId);
+            if (draft) {
+              if (draft.isValid(draftChangeListId)) {
+                var changeList = draft.getChanges();
+                explorationData.autosaveChangeList(changeList, function() {
+                  // A reload is needed so that the changelist just saved is
+                  // loaded as opposed to the exploration returned by this
+                  // explorationDataResponse.
+                  $window.location.reload();
+                });
+              } else {
+                errorCallback(explorationId, draft.getChanges());
               }
-              return response;
-            })
-          );
+            }
+            explorationDataResponse.stateTopAnswersStats =
```

Comment:
Tacking on this field like this is a bit icky. Could we make a domain object for ExplorationDataAndTopAnswerStats? It's a bit inelegant to combine the two concepts like that, but given how we're approaching this I don't see a good way around it.